R/infill_crits_interface.R

```diff
@@ -0,0 +1,366 @@
+  assertFlag(aei.use.nugget)
+  force(aei.use.nugget)
+
+  makeMBOInfillCriterion(
+    fun = function(points, models, control, par.set, design, iter, attributes = FALSE) {
+      model = models[[1L]]
+      maximize.mult = ifelse(control$minimize, 1, -1)
+      p = predict(model, newdata = points)$data
+      p.mu = maximize.mult * p$response
+      p.se = p$se
+
+      ebs = getEffectiveBestPoint(design = design, model = model, par.set = par.set, control = control)
+      # calculate EI with plugin, plugin val is mean response at ebs solution
+      d = ebs$mu - p.mu
+      xcr = d / p.se
+      xcr.prob = pnorm(xcr)
+      xcr.dens = dnorm(xcr)
+
+      # noise estimation
+      pure.noise.var = if (aei.use.nugget)
+        pure.noise.var = model$learner.model@covariance@nugget
+      else
+        estimateResidualVariance(model, data = design, target = control$y.name)
+
+      tau = sqrt(pure.noise.var)
+      res = (-1) * ifelse(p.se < 1e-06, 0,
+        (d * xcr.prob + p.se * xcr.dens) * (1 - tau / sqrt(tau^2 + p.se^2)))
+      if (attributes) {
+        res = setAttribute(res, "crit.components", data.frame(se = p$se, mean = p$response, tau = tau))
+      }
+      return(res)
+    },
+    name = "Augmeted expected improvement",
+    id = "aei",
+    components = c("se", "mean", "tau")
+  )
+}
+
+#' @export
+#' @rdname infillcrits
+makeMBOInfillCriterionEQI = function(eqi.beta = 0.75) {
+  assertNumber(eqi.beta, na.ok = FALSE, lower = 0.5, upper = 1)
+  force(eqi.beta)
+
+  makeMBOInfillCriterion(
+    fun = function(points, models, control, par.set, design, iter, attributes = FALSE) {
+      model = models[[1L]]
+      maximize.mult = ifelse(control$minimize, 1, -1)
+      # compute q.min
+      design_x = design[, (colnames(design) %nin% control$y.name)]
+      p.current.model = predict(object = model, newdata = design_x)$data
+      q.min = min(maximize.mult * p.current.model$response + qnorm(eqi.beta) * p.current.model$se)
+
+      p = predict(object = model, newdata = points)$data
+      p.mu = maximize.mult * p$response
+      p.se = p$se
+
+      pure.noise.var = if (inherits(model$learner, "regr.km")) {
+        pure.noise.var = model$learner.model@covariance@nugget
+        #FIXME: What if kriging is wrapped?
+      } else {
+        estimateResidualVariance(model, data = design, target = control$y.name)
+      }
+      tau = sqrt(pure.noise.var)
+
+      mq = p.mu + qnorm(eqi.beta) * sqrt((tau * p.se^2) / (tau + p.se^2))
+      sq = p.se^2 / sqrt(pure.noise.var + p.se^2)
+      d = q.min - mq
+      xcr = d / sq
+      xcr.prob = pnorm(xcr)
+      xcr.dens = dnorm(xcr)
+
+      eqi = ifelse(p.se < 1e-06, 0, (sq * (xcr * xcr.prob + xcr.dens)))
+      return(-eqi)
+    },
+    name = "Expected quantile improvement",
+    id = "eqi"
+  )
+}
+
+# ====================
+# MULTI-CRITERIA STUFF
+# ====================
+
+#' @export
+#' @rdname infillcrits
+makeMBOInfillCriterionDIB = function(cb.lambda = 1, cb.inflate.se = FALSE) {
```

Comment:
I am not sure if the cb.inflate.se parameter is useful for dib. The lambda value has a rather different effect here than in the CB. My suggestion: Rename to dib.lambda, delete cb.inflate.se, add dib.q from Tobias formula.